chall06/dmukaliy.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python3.7
+
+import sys
+
+def error(error_message, filename):
+    print(f'{sys.argv[0]}: {filename}: {error_message}')
+
+def get_sorted_books_width(data):
+    books = []
+    for i in range(1, len(data)):
+        book = data[i]
+        books.append(int(book.split()[0]))
+    books.sort(reverse=True)
+    return books
+
+def get_amount_shelves(shelves, books):
+    shelves_amount = -1
+    for i in range(0, len(books)):
+        found = False
+        for j in range(0, len(shelves)):
+            if shelves[j] - books[i] >= 0:
+                shelves[j] -= books[i]
+                found = True
+                break
+        if j > shelves_amount:
+            shelves_amount = j
+        if not found:
+            return -1
+    return shelves_amount + 1
+
+def parse_and_solve(data, filename):
+    shelves = [int(shelf) for shelf in data[0].split()]
+    if len(shelves) == 0:
+        exit()
+    books = get_sorted_books_width(data)
+    if sum(books) == 0:
+        print('0')
+        return
+    shelves.sort(reverse=True)
+    if shelves[0] < books[0]:
+        error('Not enough space in the given shelves', filename)
+        return
```

Comment:
yeah so if you would have use exit() you would have save a line here, and two for the function.